src/components/Text.tsx

```diff
@@ -1,59 +1,52 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import _ from 'underscore';
+import React, {ForwardedRef} from 'react';
 // eslint-disable-next-line no-restricted-imports
-import {Text as RNText} from 'react-native';
+import {Text as RNText, StyleProp} from 'react-native';
+import type {TextStyle} from 'react-native';
 import fontFamily from '../styles/fontFamily';
 import themeColors from '../styles/themes/default';
 import variables from '../styles/variables';
 
-const propTypes = {
+type TextProps = {
     /** The color of the text */
-    color: PropTypes.string,
+    color?: string;
 
     /** The size of the text */
-    fontSize: PropTypes.number,
+    fontSize?: number;
 
     /** The alignment of the text */
-    textAlign: PropTypes.string,
+    textAlign?: 'left' | 'right' | 'auto' | 'center' | 'justify' | undefined;
 
     /** Any children to display */
-    children: PropTypes.node,
+    children: React.ReactNode;
 
     /** The family of the font to use */
-    family: PropTypes.string,
+    family?: keyof typeof fontFamily;
 
     /** Any additional styles to apply */
     // eslint-disable-next-line react/forbid-prop-types
-    style: PropTypes.any,
-};
-const defaultProps = {
-    color: themeColors.text,
-    fontSize: variables.fontSizeNormal,
-    family: 'EXP_NEUE',
-    textAlign: 'left',
-    children: null,
-    style: {},
+    style?: StyleProp<TextStyle>;
```

Comment:
What if we type it `style?: TextStyle | TextStyle[];`? Then this logic seems a little easier:

```ts
    // If the style prop is an array of styles, we need to mix them all together
    const mergedStyles = !Array.isArray(style)
        ? style
        : style.reduce(
              (finalStyles, s) => ({
                  ...finalStyles,
                  ...s,
              }),
              {},
          );
    const componentStyle: TextStyle = {
        color,
        fontSize,
        textAlign,
        fontFamily: fontFamily[family],
        ...mergedStyles,
    };
```